core/src/smartcontracts/isi/world.rs

```diff
@@ -374,26 +371,26 @@ pub mod isi {
             let world = wsv.world_mut();
             let _ = world.upgraded_validator.insert(new_validator);
 
-            // let old_permission_token_definitions = wsv
-            //     .permission_token_definitions()
-            //     .values()
-            //     .cloned()
-            //     .collect::<HashSet<_>>();
-            // let new_permission_token_definitions =
-            //     HashSet::from_iter(new_permission_token_definitions);
-
-            // old_permission_token_definitions
-            //     .difference(&new_permission_token_definitions)
-            //     .map(|definition| Unregister::<PermissionTokenDefinition> {
-            //         object_id: definition.id.clone(),
-            //     })
-            //     .try_for_each(|unregister| unregister.execute(authority, wsv))?;
-
-            // new_permission_token_definitions
-            //     .difference(&old_permission_token_definitions)
-            //     .cloned()
-            //     .map(|definition| Register::<PermissionTokenDefinition> { object: definition })
-            //     .try_for_each(|new| new.execute(authority, wsv))?;
+            let old_permission_token_definitions = wsv
+                .permission_token_definitions()
+                .values()
+                .cloned()
+                .collect::<HashSet<_>>();
+            let new_permission_token_definitions =
+                HashSet::from_iter(new_permission_token_definitions);
+
+            old_permission_token_definitions
+                .difference(&new_permission_token_definitions)
+                .map(|definition| Unregister::<PermissionTokenDefinition> {
```

Comment:
maybe these should not be ISI anymore, and should be directly registered/unregistered because then you allow for our executor to also execute these operations by itself?

Reply:
also, in `client/tests/integration/permissions.rs` there is a test that tries to register a permission token. This should be disallowed, no?

Reply:
> there is a test that tries to register a permission token. This should be disallowed, no?

Yes, thank you

Reply:
> maybe these should not be ISI anymore

I'm not very sure. We probably should discuss this. But I will ban it's usage by our defalut validator

Reply:
> I'm not very sure

Can you comment on why this operation should be allowed to be executed by the user? Because anyone will be able to execute this operation from the validator

Reply:
> > there is a test that tries to register a permission token. This should be disallowed, no?
> 
> Yes, thank you

Turns out these are ignored tests. Ignored for a long time and I don't why. We should revisit #2581. I'm not sure if I should dig into these tests in this PR. I think we can find more outdated stuff in there

Reply:
I think if all permission tokens are controlled by `validator` there is no point in preserving `Register/Unregister<PermissionToken>`. It will just add additional surface for vulnerabilities.

Reply:
I think that this should be removed. But I don't think that it necessarily needs to be removed in this PR.

Reply:
since there is a concern about modifying `genesis.json` @Arjentix you mark these two methods as deprecated and have them `panic`/return error if somebody calls them accidentally

Reply:
Removed these instructions